### PR TITLE
Buffer vim microstate

### PR DESCRIPTION
**Pull request by Dietr1ch**
#4778
Adds j/k bindings to move around buffers on the buffer-transient-state. Old bindings remain.
This adds some consistency for vim users that prefer j/k instead of n/p or n/N.

### CODE REVIEW

File: layers/+lang/d/README.org [hunk 1]
```diff
@@ -30,3 +31,6 @@ instructions]]. After successfully built DCD, you need to copy the binary
 | ~SPC m g r~ | find references to all symbol at point                |
 |-------------+-------------------------------------------------------|
 
+* Requirements
+	- [[https://github.com/Hackerpilot/DCD][DCD]] (completion)
+	- [[https://github.com/dlang/dub][DUB]] (dub projects)
```
Comment (nixmaniack) [inline layers/+lang/d/README.org]: These changes should be part of another PR.

**Comment by nixmaniack** [comment]
IMO, this would break the spacemacs convention(`n/N/p`) which is followed across transient states and may also start conflicting if we're using `j/k` in transients states already.

**Comment by rgrinberg** [comment]
I don't think j/k are consistent with vim here either. So I would rather not have this. Perhaps this could be a preference for users though?

**Comment by Dietr1ch** [comment]
I agree that using `n/j` and `n/N/k` for moving seems bloated, but having the bindings will only give you more choices.
While `j/k` are not used for `:bnext` and `:bprev` on vim, many vim-inspired bindings do use `j/k` for moving through options.
Otherwise, using `j/k` by mistake will only take you out of the transient state.


If you search for `"[jk]"` on spacemacs, many bindings appear. I wouldn't say that `n/Np` is really established over `j/k`. Preference of `n/p` over `j/k` seems to vary depending on which one caused less conflicts with the other bindings.


If you are really sure that `n/Np` is closer to be default on spacemacs I'll close this and eventually get used to it, but I kept trying to use `j/k` on this transient state and thought that I was not the only one.

**Comment by nixmaniack** [comment]
> but having the bindings will only give you more choices.

IMHO, having set of conventions will be more beneficial than having more choices as number of keybindings are going to increase. 

[Here's the link](https://github.com/syl20bnr/spacemacs/blob/aa3a799c39afab721ab47a1d24294d6a753c9e54/doc/CONVENTIONS.org#n-and-n) to spacemacs' convention which mentions about usage of `n/N`.

May be this could be considered as a proposal to re-think about that convention then and move to using `j/k` in TS? I'm weighing towards `n/N/p` as it indicates `next/previous` (may be I'm biased as I'm used to it!) more clearly than `j/k` (which I think of as `up/down`).

**Comment by Dietr1ch** [comment]
Yes, it's true that `j/k` suggests vertical movement, and that `n/N/p` does not, while also being a little vim-ish.  It seems that I'l have to get used to it to favor consistency, eventually it'll be easier.
Thanks for your time (=
